lightning/src/ln/channelmanager.rs

```diff
@@ -1333,6 +1339,39 @@ impl<ChanSigner: ChannelKeys, M: Deref, T: Deref, K: Deref, F: Deref> ChannelMan
 		})
 	}
 
+	/// Generates a signed node_announcement from the given arguments and creates a
+	/// BroadcastNodeAnnouncement event.
+	///
+	/// RGB is a node "color" and alias ia a printable human-readable string to describe this node
+	/// to humans. They carry no in-protocol meaning.
+	///
+	/// addresses represent the set (possibly empty) of socket addresses on which this node accepts
+	/// incoming connections. These will be broadcast to the network, publicly tying these
+	/// addresses together. If you wish to preserve user privacy, addresses should likely contain
+	/// only Tor Onion addresses.
+	pub fn broadcast_node_announcement(&self, rgb: [u8; 3], alias: [u8; 32], addresses: msgs::NetAddressSet) {
+		let _ = self.total_consistency_lock.read().unwrap();
+
+		let announcement = msgs::UnsignedNodeAnnouncement {
+			features: NodeFeatures::supported(),
+			timestamp: self.last_node_announcement_serial.fetch_add(1, Ordering::AcqRel) as u32,
```

Comment:
reasoning for `AcqRel`?

Reply:
`AcqRel` is a good default - it ensures we always have the latest value here, without acting as a lock (as `SeqCst` does). If we're not relying on any other data to be consistent, but want consistency ourselves, `AcqRel` it is. Note that, on x86, `AcqRel` compiles down to nothing, whereas `SeqCst` is relatively expensive.